### ISSUE TITLE
loss function

### ISSUE THREAD

**Issue by holyseven** [MelodyHai/caffe-yolo#3]
Hi MelodyHai,

thanks for sharing your codes with us. 

After reading some .prototxt in your [example/yolo](https://github.com/MelodyHai/caffe-yolo/tree/master/examples/yolo), I have the feeling that you use softmax loss for classification and Euclidean loss for coordinates regression, and treat them equally (without multiplying penalty factors). However, in the article yolo, the author said that this "may not be ideal" and of course, the author uses sum-squared error for classification but I think that using softmax loss is fine. 

The important issue is the penalty factors between classification error and coordinates error, and the factors between grids where have object(s) in them and grids where have no objects. I didn't find those factors in your codes. Did I miss them?

Have you trained the yolo model with your codes? Does it work well?

Thanks,
Jacques.

**Comment by killogre** [comment]
I agree with @holyseven: the lambda_coord parameter is missing - both regression and classification terms get the same weight. Also, both the position (x,y) and size (w,h) get the same treatment, whereas in the paper the authors suggest taking the square root of the size parameters.
Can you please comment on that, @MelodyHai  ?
